types/index.d.ts

```diff
@@ -44,6 +44,9 @@ declare module "skedify-sdk" {
           cb: (filterItem: WithFilter<Entity, AdditionalFilters>) => unknown
         ) => GenericAPI<Entity>;
 
+    page(pageNo: number): GenericAPI<Entity>;
+    limit(limit: number): GenericAPI<Entity>;
```

Comment:
Slight typing improvement possible here to avoid the always optional:

Define a helper at the toplevel of this file (under `type WithDelete<T, Response>`)
```ts
  type GenericAPIWithPaging<Entity> = GenericAPI<Entity, APIResponse<Entity> & { paging: PagingResponse }>;
```

```suggestion
    page(limit: number): {
      limit(pageNo: number): GenericAPIWithPaging<Entity>
    };

    limit(limit: number): {
      page(pageNo: number): GenericAPIWithPaging<Entity>
    };
```

This will require you to call `page().limit()` or `limit().page()` in succession when using one of the 2.

And we can then explicitly add on the paging response after the consumer has called the 2nd argument